src/test/regress/sql/grant_on_schema_propagation.sql

```diff
@@ -0,0 +1,178 @@
+GRANT CREATE ON SCHEMA grantor_schema TO role_2;
+GRANT USAGE, CREATE ON SCHEMA grantor_schema TO role_3;
+GRANT CREATE, USAGE ON SCHEMA grantor_schema TO PUBLIC;
+SET ROLE role_2;
+GRANT USAGE ON SCHEMA grantor_schema TO role_3;
+RESET ROLE;
+
+-- distribute the schema
+CREATE TABLE grantor_schema.grantor_table (id INT);
+SELECT create_distributed_table('grantor_schema.grantor_table', 'id');
+
+-- check if the grantors are propagated correctly
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname = 'grantor_schema' ORDER BY nspname;
+\c - - - :worker_1_port
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname = 'grantor_schema' ORDER BY nspname;
+\c - - - :master_port
+
+-- add the previously removed node
+SELECT 1 FROM master_add_node('localhost', :worker_2_port);
+
+-- check if the grantors are propagated correctly
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname = 'grantor_schema' ORDER BY nspname;
+\c - - - :worker_2_port
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname = 'grantor_schema' ORDER BY nspname;
+\c - - - :master_port
+
+-- revoke one of the permissions
+REVOKE USAGE ON SCHEMA grantor_schema FROM role_1 CASCADE;
+
+-- check if revoke worked correctly
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname = 'grantor_schema' ORDER BY nspname;
+\c - - - :worker_1_port
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname = 'grantor_schema' ORDER BY nspname;
+\c - - - :master_port
+
+-- test if grantor propagates correctly on already distributed schemas
+GRANT USAGE ON SCHEMA grantor_schema TO role_1 WITH GRANT OPTION;
+SET ROLE role_1;
+GRANT USAGE ON SCHEMA grantor_schema TO role_2;
+GRANT USAGE ON SCHEMA grantor_schema TO role_3 WITH GRANT OPTION;
+SET ROLE role_3;
+GRANT USAGE ON SCHEMA grantor_schema TO role_2;
+RESET ROLE;
+
+-- check the results
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname = 'grantor_schema' ORDER BY nspname;
+\c - - - :worker_1_port
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname = 'grantor_schema' ORDER BY nspname;
+\c - - - :master_port
+
+DROP TABLE grantor_schema.grantor_table;
+SELECT run_command_on_coordinator_and_workers('DROP SCHEMA grantor_schema CASCADE');
+
+-- test distributing the schema with another user
+CREATE SCHEMA dist_schema;
+
+GRANT ALL ON SCHEMA dist_schema TO role_1 WITH GRANT OPTION;
+SET ROLE role_1;
+GRANT ALL ON SCHEMA dist_schema TO role_2 WITH GRANT OPTION;
+
+CREATE TABLE dist_schema.dist_table (id int);
+SELECT create_distributed_table('dist_schema.dist_table', 'id');
+
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname = 'dist_schema' ORDER BY nspname;
+\c - - - :worker_1_port
+SELECT nspname, nspacl FROM pg_namespace WHERE nspname = 'dist_schema' ORDER BY nspname;
+\c - - - :master_port
+
+DROP TABLE dist_schema.dist_table;
+SELECT run_command_on_coordinator_and_workers('DROP SCHEMA dist_schema CASCADE');
+SELECT run_command_on_coordinator_and_workers('DROP ROLE role_1, role_2, role_3');
+SET citus.shard_replication_factor TO 2;
```

Comment:
this is not required. Every test is ran in their own session which causes this variable to rest after the test before the next test.